pkg/security/ImageScanService.go

```diff
@@ -373,6 +378,17 @@ func (impl ImageScanServiceImpl) FetchExecutionDetailResult(request *ImageScanRe
 			}
 			executionTime = item.ImageScanExecutionHistory.ExecutionTime
 		}
+		if len(imageScanResult) > 0 {
+			imageScanResponse.ScanToolId = imageScanResult[0].ScanToolId
+		} else {
+			scanToolHistoryMappings, err := impl.scanToolExecutionHistoryRepository.GetAllScanHistoriesByExecutionHistoryIds(scanExecutionIds)
+			if err != nil {
```

Comment:
why are we not returning from here? we might get connection error or something else. If empty result is okay, then add check for NoRows and return for remaining errors